recipes/wavelet_buffer/all/conanfile.py

```diff
@@ -0,0 +1,177 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake, cmake_layout
+from conan.tools.files import get, copy, rmdir
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+
+import os
+
+required_conan_version = ">=1.50"
+
+
+class WaveletBufferConan(ConanFile):
+    name = "wavelet_buffer"
+    license = "MPL-2.0"
+    description = "An universal C++ compression library based on wavelet transformation"
+    topics = ("compression", "signal-processing", "wavelet")
+    homepage = "https://github.com/panda-official/WaveletBuffer"
+    url = "https://github.com/conan-io/conan-center-index"
+    default_options = {
+        "cimg/*:enable_fftw": False,
```

Comment:
This options in "On" mode will demand heavy, conflicting and extra dependencies. Which wavelet buffer doesn't need.
According to documentation `dependencies` interface could be used only in `validate()` and `generate()` method in readonly mode.
So I put it in `default_options`

Reply:
don't do that please

Reply:
@SpaceIm ok, How to do it properly? How to set them to build conan package with options value which I want?

Reply:
You set this options in root user recipe.

Reply:
@uilianries  @SpaceIm  Try to hear me, pls: wavelet_buffer does need CImg with that options set. Due default options set will create dependency conflict, moreover extra heavy packages will be installed which  are not needed for wavelet_buffer

Reply:
@victor1234 It's not allowed by CCI. You can added restriction rules to `validate()` method, but you can not force on that way.

Reply:
@uilianries this means that all CI-provided configuration will be rejected. And conan-index repo will only contain package recipe without binaries. And every user will have to set options manually and compile package locally.
I get it. But it nullifies sense of using public repo.

Reply:
No, there is currently a dependency graph conflict in cimg recipe, but it will be fixed by https://github.com/conan-io/conan-center-index/pull/13917

Reply:
@SpaceIm i'm completely lost. What should I do to complete PR?

Reply:
Wait https://github.com/conan-io/conan-center-index/pull/13917 to be merged, which requires https://github.com/conan-io/conan-center-index/pull/13406, which requires https://github.com/conan-io/conan-center-index/pull/13285, which requires https://github.com/conan-io/conan-center-index/issues/13159 to be fixed.

Reply:
Otherwise we could change default options of cimg recipe. Since it's a header_only recipe, it's harmless.

Reply:
I've disabled all options by default in https://github.com/conan-io/conan-center-index/pull/13917. Since cimg is header_only, it doesn't matter in terms of package id.